# Hypothesis Testing Problems

# Population Mean, σ Known

Ques - Individuals filing federal income tax returns prior to March 31 received an average refund of $1056. Consider the population of “last-minute” filers who mail their tax return during the last five days of the income tax period (typically April 10 to April 15).

a. A researcher suggests that a reason individuals wait until the last five days is that on average these individuals receive lower refunds than do early filers. Develop appropriate hypotheses such that rejection of H0 will support the researcher’s contention.

b. For a sample of 400 individuals who filed a tax return between April 10 and 15, the
sample mean refund was $910. Based on prior experience a population standard deviation of σ = $1600 may be assumed. What is the p-value?

c. At α = .05, what is your conclusion?


    Ans (a) 
    Null Hypothesis Population Mean >= 1056

    Alternative Hypthesis  Population Mean < 1056

    It is a Lower Tail Test

In [2]:
# Ans (b)
# Create a Function to Calculate Z value
# x= 910, mu = 1056, sd = 1600, n = 400

from scipy import stats
import math

def Value(x, mu, sd, n):
    sqRoot = math.pow(n, 0.5)
    Deno = sd/sqRoot
    return (x-mu)/Deno
 
z_Value = Value(910, 1056, 1600, 400)
print(z_Value)

# Calculate P-Value
stats.norm.cdf(z_Value)  # Lower Tail Test

-1.825


0.03400051466982242

    Ans (b) - P value = 0.03400051466982242

    Ans (c) - As P-Value is less than α

    Therefore, Reject the Null Hypothesis.

    To conclude, that the mean refund of “last minute” filers is less than $1056

Ques - CN and ActMedia provided a television channel targeted to individuals waiting in supermarket checkout lines. The channel showed news, short features, and advertisements. The length of the program was based on the assumption that the population mean time a shopper stands in a supermarket checkout line is 8 minutes. A sample of actual waiting times
will be used to test this assumption and determine whether actual mean waiting time differs from this standard.

a. Formulate the hypotheses for this application.

b. A sample of 120 shoppers showed a sample mean waiting time of 7.5 minutes. Assume a population standard deviation σ = 3.2 minutes. What is the p-value?

c. At α = .05, what is your conclusion?

    Ans (a) 
    Null Hypothesis, Population Mean = 8 (Equal)

    Alternative Hypthesis, Population Mean <> 8 (Not equal)

    It is a Two Tail Test

In [2]:
# Ans(b), Use Previous Function to Calculate Z Value
# x= 7.5, mu = 8, sd = 3.2, n = 120 
z_Value = Value(7.5, 8, 3.2, 120)
print(z_Value)

# Calculate P-Value
# for Two Tail Test 
2 * stats.norm.cdf(z_Value)  

-1.7116329922036442


0.08696432431499454

    Ans (b) - P value = 0.08696432431499454

    Ans (c) - As P-Value is Greater than α

    Therefore, Do not Reject the Null Hypothesis.

    cannot conclude the mean waiting time differs from eight minutes.

# Population Mean, σ Unknown

Ques - Average Sales of Ice cream in a local Shop of Delhi for last month was 10 . Shopowner Anticipated that sample data would show evidence that mean sales of ice cream below the the 10.

a. Develop hypothesis Test

b. Calculate P-value

c. At α = .05, what is your conclusion?

Data for Sample fetch from Excel

In [3]:
# Import Library
import pandas as pd

In [4]:
Sales = pd.read_csv('C:\\Users\\JITENDRA\\Downloads\\Data for Stat Projects\\Ice- Cream Sales.csv')
Sales.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20 entries, 0 to 19
Data columns (total 2 columns):
 #   Column                 Non-Null Count  Dtype
---  ------                 --------------  -----
 0   Day                    20 non-null     int64
 1   No. of Ice Cream Sold  20 non-null     int64
dtypes: int64(2)
memory usage: 448.0 bytes


In [5]:
Sales.head()

Day   No. of Ice Cream Sold
0     1                     13
1     2                      8
2     3                     10
3     4                     10
4     5                      8

    Ans(a) 

    Null Hypothesis, Population Mean <= 10

    Alternative Hypthesis, Population Mean > 10 

    It is a Upper Tail Test

In [6]:
# Ans(b), Calculate P-Value
# convert Dataframe to list
col_one_list = Sales['No. of Ice Cream Sold'].tolist()
col_one_list
print(stats.ttest_1samp(col_one_list, 10))   # This give P-value for Two Tailed test

# P-values for Upper Tail test
P_value = 0.7239703579964252/2    # Result of Two Tail test/2
print(P_value)


Ttest_1sampResult(statistic=-0.35843385854878496, pvalue=0.7239703579964252)
0.3619851789982126


    Ans (b) - P value = 0.3619851789982126

    Ans(c) 

    As P-Value is Greater than α

    Therefore, Accept the Null Hypothesis.

    To conclude the sales of Ice-cream is below 10

# Population Proportion

Ques -  study by Consumer Reports showed that 64% of supermarket shoppers believe supermarket brands to be as good as national name brands. To investigate whether this result applies to its own product, the manufacturer of a national name-brand ketchup asked a sampleof shoppers whether they believed that supermarket ketchup was as good as the national
brand ketchup.

a. Formulate the hypotheses that could be used to determine whether the percentage of supermarket shoppers who believe that the supermarket ketchup was as good as the
national brand ketchup differed from 64%.

b. If a sample of 100 shoppers showed 52 stating that the supermarket brand was as good as the national brand, what is the p-value?

c. At α = .05, what is your conclusion?

d. Should the national brand ketchup manufacturer be pleased with this conclusion?
Explain

    Ans(a) 

    Null Hypothesis, Population Proportion >= 0.64

    Alternative Hypthesis, Population Mean < 0.64

    It is a Lower Tail Test

In [14]:
# Ans(b)
# Import Proportion Library
from statsmodels.stats.proportion import proportions_ztest
Count = 52
samplesize = 100
p = 0.64
print(proportions_ztest(Count, samplesize, p))  # Giving P_value for Two tailed Test
P_value = 0.016309171877754974/2      # Gives P_Value for Lower Tailed Test 
print(P_value)

(-2.401922307076307, 0.016309171877754974)
0.008154585938877487


    Ans (b) - P value = 0.008154585938877487

    Ans(c) - As P-Value is less than α. Therefore, Reject the Null Hypothesis.

    To conclude, less than 64% of shoppers believe supermarket ketchup is as good as the
    national brand

Ans(d) 

Yes, National brand Ketchup Manufacturer must be Happy with conclusion

# Calculation of Type 1 and Type 2 Error

Ques - A consumer research group is interested in testing an automobile manufacturer’s claim that a new economy model will travel at least 25 miles per gallon of gasoline 
(H0: µ >= 25).

a. Calculate Type 1 error.

b. With a .02 level of significance and a sample of 30 cars, what is the rejection rule based on the value of for the test to determine whether the manufacturer’s claim should be
rejected? Assume that σ is 3 miles per gallon.

c. What is the probability of committing a Type II error if the actual mileage is 23 miles
per gallon?

d. What is the probability of committing a Type II error if the actual mileage is 24 miles
per gallon?

e. What is the probability of committing a Type II error if the actual mileage is 25.5 miles
per gallon?

    Ans (a)
    Type 1 error occurs when the null hypothesis is true but we reject it.

    Mathematically, Type 1 Error = Level of Significance
    
    Therefore, for this Problem, Type 1 error = 2 percent

In [3]:
# Ans (b), I need to calculate X bar
# Calculation for X bar value
#  
def XValue( mu, sd, n):
    sqRoot = math.pow(n, 0.5)
    Deno = sd/sqRoot
    z = stats.norm.ppf(0.02)
    Z_valuetype = z*Deno
    return mu + Z_valuetype

# calling a function
XValue(25, 3, 30 )   

23.87511539419529

    Ans (b)

    Decision Rule

    Accept H0 if X bar >= 23.88

    Reject H0 if X bar < 23.88

In [14]:
# Creating Function to Calculate Type 2 Error
import numpy as np
def type_2(mu1,mu2,sigma,n,alfa):
    z = stats.norm.ppf(alfa)
    xbar = (mu1)+(z*sigma/np.sqrt(n))
    z2 = (xbar - mu2)/(sigma/np.sqrt(n))
    if(mu1>mu2):
        beta = 1-stats.norm.cdf(z2)
    else:
        beta = stats.norm.cdf(z2)
    
    print(beta)


    Ans (c)

    Type 2 error (Beta) = 5.5 percent


In [16]:
# Ans(c), type 2 error (Beta) for mileage is equal to 24 miles per gallon
type_2(25,24,3,30,0.02)

0.5901796210049072


    Ans (d)

    Type 2 error (Beta) = 59 percent

In [20]:
#Ans(d), type 2 error (Beta) for mileage is equal to 24 miles per gallon
# Calling Z values 
type_2(25,25.5,3,30,0.02) 

0.0015054649762203938


    Ans (e)

    Type 2 error (Beta) = 0.1 percent

    Conclusion 

    Shows, the probability of a Type II error depends on the value of the population mean µ. For values of µ near µ0, the probability of making the Type II error can be high.